### PR TITLE
handling additional keywords

### PR DESCRIPTION
**Pull request by MahmoudAlamar03**
Make sure additional keywords are handled for TOLPV (Nexus)

### CODE REVIEW

File: ResSimpy/Nexus/DataModels/StructuredGrid/NexusGrid.py [hunk 1]
```diff
@@ -603,17 +603,36 @@ def move_next_value(next_line: str) -> tuple[str, str]:
...
+                        try:
+                            float(first_value)
+                        except ValueError:
+                            selector = first_value.upper()
+                            value_index = 2
+
+                    self._tolpv = float(tolpv_values[value_index])
+                    self._tolpv_grid_name = selector
+
+                    if selector is not None:
```
Comment (yzaparb) [inline ResSimpy/Nexus/DataModels/StructuredGrid/NexusGrid.py:630]: Great to have this warning!

**Review by yzaparb — APPROVED** [review]
LGTM!

File: ResSimpy/Nexus/DataModels/StructuredGrid/NexusGrid.py [hunk 2]
```diff
@@ -603,17 +603,36 @@ def move_next_value(next_line: str) -> tuple[str, str]:
...
                 self._range_z = int(third_value)
 
-            # Load TOLPV scalar value
+            # Load TOLPV scalar value, optionally with a grid selector (e.g. TOLPV LGR1 0 or TOLPV ALL 0).
             if nfo.check_token('TOLPV', line):
-                tolpv_value = fo.get_token_value('TOLPV', line, [line])
-                if tolpv_value is not None:
-                    self._tolpv = float(tolpv_value)
+                tolpv_values = fo.split_line(line, upper=False)
+                if len(tolpv_values) >= 2 and tolpv_values[0].upper() == 'TOLPV':
```
Comment (cflynn3) [inline ResSimpy/Nexus/DataModels/StructuredGrid/NexusGrid.py:609]: Do we want a warning / error if this isn't the case?

File: ResSimpy/Nexus/DataModels/StructuredGrid/NexusGrid.py [hunk 3]
```diff
@@ -603,17 +603,36 @@ def move_next_value(next_line: str) -> tuple[str, str]:
                 self._range_y = int(second_value)
                 self._range_z = int(third_value)
 
-            # Load TOLPV scalar value
+            # Load TOLPV scalar value, optionally with a grid selector (e.g. TOLPV LGR1 0 or TOLPV ALL 0).
             if nfo.check_token('TOLPV', line):
-                tolpv_value = fo.get_token_value('TOLPV', line, [line])
-                if tolpv_value is not None:
-                    self._tolpv = float(tolpv_value)
+                tolpv_values = fo.split_line(line, upper=False)
```
Comment (cflynn3) [inline ResSimpy/Nexus/DataModels/StructuredGrid/NexusGrid.py:608]: What happens if we have the value on the next line?

Comment (cflynn3) [reply ResSimpy/Nexus/DataModels/StructuredGrid/NexusGrid.py:608]: Or comments after values etc (e.g. TOLPV ALL !this is a tolpv line). Please add these as test cases below before trying to fix anything.

File: ResSimpy/Nexus/DataModels/StructuredGrid/NexusGrid.py [hunk 4]
```diff
@@ -603,17 +603,36 @@ def move_next_value(next_line: str) -> tuple[str, str]:
...
+                    selector: str | None = None
+                    value_index = 1
+
+                    if len(tolpv_values) >= 3:
+                        first_value = tolpv_values[1]
+                        try:
+                            float(first_value)
+                        except ValueError:
+                            selector = first_value.upper()
+                            value_index = 2
```
Comment (cflynn3) [inline ResSimpy/Nexus/DataModels/StructuredGrid/NexusGrid.py]: I feel like `get_nth_value()` would do a better job that this code splitting the line and trying to parse the values.

File: tests/Nexus/test_nexus_tolpv.py [hunk 5]
```diff
@@ -65,3 +67,70 @@ def mock_open_wrapper(filename, mode):
...
+    # Act
+    simulation = NexusSimulator(origin='testpath1/nexus_run.fcs')
+    assert simulation.grid is not None
+    with pytest.warns(UserWarning, match=r'TOLPV in Nexus was applied to grid ALL'):
+        result = simulation.grid.tolpv
+
+    # assert
+    assert isinstance(result, float)
+    assert result == 5000.5
+    assert simulation.grid.tolpv_grid_name == 'ALL'
```
Comment (cflynn3) [inline tests/Nexus/test_nexus_tolpv.py:137]: These tests are good, but as mentioned above, please expand them out to include more edge cases.

**Review by cflynn3 — COMMENTED** [review]
It looks OK for perfect models, but please consider the edge cases I mention.

File: file.py [hunk 6]
```diff
@@ -0,0 +1,2 @@
+import sys
+print(sys.executable)
```
Comment (obi-isebor) [inline file.py]: What is this needed for?

**Review by obi-isebor — APPROVED** [review]
Please see comment on file.py, otherwise, LGTM

File: file.py (hunk 6 above)
Comment (MahmoudAlamar03) [reply file.py]: oops, this wasn't supposed to be committed.
